polynote-frontend/polynote/ui/component/notebook/notebook.ts

```diff
@@ -86,9 +91,13 @@ export class Notebook extends Disposable {
                     // index not found, must be at the end
                     cellsEl.appendChild(el);
                 }
+
+                if (layoutNewCells) {
+                    cell.layout()
+                }
             })
 
-            if (needLayout) {
+            if (layoutAllCells) {
```

Comment:
When I was testing this PR it looked like the layout that gets called by this may not be necessary any more, perhaps due to the notebook loaded Promise change? I'm not sure, but from what I could tell at this point the cells aren't in the DOM yet. 

I left it in because I wasn't sure if there was an edge case I missed.